store/tikv/gcworker/gc_worker.go

```diff
@@ -1659,28 +1692,34 @@ func (w *GCWorker) saveValueToSysTable(key, value string) error {
 }
 
 // RunGCJob sends GC command to KV. It is exported for kv api, do not use it with GCWorker at the same time.
-func RunGCJob(ctx context.Context, s tikv.Storage, safePoint uint64, identifier string, concurrency int) error {
+func RunGCJob(ctx context.Context, s tikv.Storage, pd pd.Client, safePoint uint64, identifier string, concurrency int) error {
 	gcWorker := &GCWorker{
-		store: s,
-		uuid:  identifier,
+		store:    s,
+		uuid:     identifier,
+		pdClient: pd,
+	}
+
+	if concurrency <= 0 {
+		return errors.Errorf("[gc worker] gc concurrency should greater than 0, current concurrency: %v", concurrency)
 	}
 
 	err := gcWorker.resolveLocks(ctx, safePoint, concurrency, false)
 	if err != nil {
 		return errors.Trace(err)
 	}
 
-	if concurrency <= 0 {
-		return errors.Errorf("[gc worker] gc concurrency should greater than 0, current concurrency: %v", concurrency)
-	}
-
 	err = gcWorker.saveSafePoint(gcWorker.store.GetSafePointKV(), safePoint)
 	if err != nil {
 		return errors.Trace(err)
 	}
 	// Sleep to wait for all other tidb instances update their safepoint cache.
 	time.Sleep(gcSafePointCacheInterval)
 
+	safePoint, err = gcWorker.setServiceSafePoint(ctx, safePoint)
```

Comment:
should we set it in `calculateNewSafePoint`? then it will be consistent with the value saved in TiKV, or should we remove the data of safepoint in TiKV?

Reply:
Actually I'm not sure how to do it is better. I thought other safe points don't need to block resovleLock phase. But now I found that it introduces too much complexity. I'll move it to calculateNewSafePoint.

Reply:
I think `UpdateServiceGCSafePoint` should be a replacement of `saveSafePoint`, so can `saveSafePoint` be removed here?

Reply:
No. It needs more modifications to remove it. We can do it later, not in this PR which means to quickly make it possible for CDC to pause the GC.